src/java.base/share/classes/jdk/internal/foreign/abi/ppc64/aix/AixPPC64Linker.java

```diff
@@ -59,23 +59,23 @@ public static AixPPC64Linker getInstance() {
     }
 
     @Override
-    protected MethodHandle arrangeDowncall(MethodType inferredMethodType, FunctionDescriptor function) {
-        return CallArranger.arrangeDowncall(inferredMethodType, function);
+    protected MethodHandle arrangeDowncall(MethodType inferredMethodType, FunctionDescriptor function, LinkerOptions options) {
+        return CallArranger.arrangeDowncall(inferredMethodType, function, options);
     }
 
     @Override
-    protected MemorySegment arrangeUpcall(MethodHandle target, MethodType targetType, FunctionDescriptor function, MemorySession session) {
-        return CallArranger.arrangeUpcall(target, targetType, function, session);
+    protected MemorySegment arrangeUpcall(MethodHandle target, MethodType targetType, FunctionDescriptor function, SegmentScope scope) {
+        return CallArranger.arrangeUpcall(target, targetType, function, scope);
     }
 
-    public static VaList newVaList(Consumer<VaList.Builder> actions, MemorySession session) {
-        AixPPC64VaList.Builder builder = AixPPC64VaList.builder(session);
+    public static VaList newVaList(Consumer<VaList.Builder> actions, SegmentScope scope) {
+        AixPPC64VaList.Builder builder = AixPPC64VaList.builder(scope);
         actions.accept(builder);
         return builder.build();
     }
 
-    public static VaList newVaListOfAddress(MemoryAddress ma, MemorySession session) {
-        return AixPPC64VaList.ofAddress(ma, session);
+    public static VaList newVaListOfAddress(long addr, SegmentScope scope) {
+        return AixPPC64VaList.ofAddress(addr, scope);
```

Comment:
Please use the full word `address` for the first parameter. Same comment for `SysVPPC64leLinker` and `SysVS390xLinker`.

Reply:
Renamed to `address` as suggested above.

Please note we don't have to all naming conventions of OpenJDK everywhere given the VaList APIs on Power & zLinux were totally implemented by ourselves.